### PR TITLE
Update to 4.7.0

### PR DESCRIPTION
**Pull request by jasongrout**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jasongrout** [comment]
Can we use noarch? Trying...

@conda-forge-admin, please add noarch: python

**Comment by bollwyvl** [comment]
We can only go noarch (which would be awesome) if we actually drop pywin32... looking at the source...

**Comment by jasongrout** [comment]
> We can only go noarch (which would be awesome) if we actually drop pywin32... looking at the source...

According to the conda-forge guidelines at https://conda-forge.org/docs/maintainer/knowledge_base.html?#noarch-python:

> noarch is a statement about the package’s source code and not its install environment. A package is still considered noarch even if one of its dependencies is not available on a given platform. If this is the case, conda will display a helpful error message describing which dependency couldn’t be found when it tries to install the package. If the dependency is later made available, your package will be installable on that platform without having to make any changes to the feedstock. However, keep in mind that since noarch packages are built on Linux, all dependencies must be available on Linux. 

Oh, wait, pywin32 is not available on linux...

**Comment by bollwyvl** [comment]
Whether or not it's used (which can be hard to find, in my experience), it looks like it's still in the upstream [setup](https://github.com/jupyter/jupyter_core/blob/master/setup.cfg#L25-L27) which would run afoul.

I see one definite reference in the tests to [windll](https://github.com/jupyter/jupyter_core/blob/5894f2e64c3182aa176d0ebb5563cb88c81bf351/jupyter_core/tests/test_paths.py#L266).

**Comment by jasongrout** [comment]
Thanks. I took out some of the test skipping that you put in a while ago, since it seemed that should be temporary.

**Comment by bollwyvl** [comment]
Oooh, yeah, for sure, can't get rid of pywin32 [yet](https://github.com/jupyter/jupyter_core/blob/0870271a25fede7a37f4f6932d38293e6cc5c04b/jupyter_core/paths.py#L387-L389).

File: recipe/meta.yaml
Comment (bollwyvl) [inline recipe/meta.yaml:31]: would be great! it just sometimes, tests don't plan for the "fun" of running in n layers of docker, cross-device links, etc.

File: recipe/meta.yaml
Comment (bollwyvl) [inline recipe/meta.yaml:42]: :+1: i've also recently been adding `--pytest-cov` and a `--cov-fail-under` (set to whatever it is) just to backstop pip check missing a major possible test dependency...

Comment (jasongrout) [reply recipe/meta.yaml:42]: Great, thanks for the tips. I'm in unfamiliar territory here, so appreciate all the help!

Comment (jasongrout) [reply recipe/meta.yaml:42]: >  (set to whatever it is

Not sure "whatever it is" is. Do you want to submit a code suggestion, just to be clear?

I'm also fine to abort these testing changes and go back to what works...

Comment (bollwyvl) [reply recipe/meta.yaml]: Sorry, I meant whatever (the threshold happens to be today) e.g. `76` on linux. This may well be a bridge too far, and can be added at another point. I like this as a prelude to turning on automerge, which otherwise gives me the willies for important packages. But you have the stuff working, and can do another day, if you'd rather we :shipit: !

Anyhoo: for posterity

```diff
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index f231d31..06168a2 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -29,19 +29,21 @@ requirements:
     - pywin32  # [win]
     - traitlets
 
-{% set skip = "not test_not_on_path and not test_path_priority" %}
+{% set skip = "not (not_on_path or path_priority)" %}
+
 test:
   requires:
     - ipykernel
     - nose
-    - pytest
     - pip
+    - pytest
+    - pytest-cov
   commands:
+    - pip check
     - jupyter -h
     - jupyter-migrate -h
     - jupyter-troubleshoot --help
-    - python -m pytest --pyargs jupyter_core -k "{{ skip }}"
-    - pip check
+    - python -m pytest -vv --pyargs jupyter_core -k "{{ skip }}" --cov jupyter_core --cov-fail-under 76
   imports:
     - jupyter_core
 

```

**Review by bollwyvl — APPROVED** [review]
lgtm, if you don't want to keep fiddling with the tests... thanks so much for turning them back on!

**Comment by bollwyvl** [comment]
Here's what showed up: 
```
linux-ppc64le/jupyter_core-4.7.0-py39hc1b9086_0.tar.bz2     # maybe some missing?
linux-64/jupyter_core-4.7.0-py36h5fab9bb_0.tar.bz2
linux-64/jupyter_core-4.7.0-py36hd000896_0.tar.bz2
linux-64/jupyter_core-4.7.0-py37h89c1867_0.tar.bz2
linux-64/jupyter_core-4.7.0-py38h578d9bd_0.tar.bz2
linux-64/jupyter_core-4.7.0-py39hf3d152e_0.tar.bz2 
linux-aarch64/jupyter_core-4.7.0-py36h2e20a7f_0.tar.bz2
linux-aarch64/jupyter_core-4.7.0-py36h704843e_0.tar.bz2
linux-aarch64/jupyter_core-4.7.0-py37hd9ded2f_0.tar.bz2
linux-aarch64/jupyter_core-4.7.0-py38h2063c64_0.tar.bz2
linux-aarch64/jupyter_core-4.7.0-py39ha65689a_0.tar.bz2
linux-ppc64le/jupyter_core-4.7.0-py36h270354c_0.tar.bz2
linux-ppc64le/jupyter_core-4.7.0-py36he7cccf2_0.tar.bz2
linux-ppc64le/jupyter_core-4.7.0-py37h35e4cab_0.tar.bz2
linux-ppc64le/jupyter_core-4.7.0-py38hf8b3453_0.tar.bz2
osx-64/jupyter_core-4.7.0-py36h5dafb3c_0.tar.bz2
osx-64/jupyter_core-4.7.0-py36h79c6626_0.tar.bz2
osx-64/jupyter_core-4.7.0-py37hf985489_0.tar.bz2
osx-64/jupyter_core-4.7.0-py38h50d1736_0.tar.bz2
osx-64/jupyter_core-4.7.0-py39h6e9494a_0.tar.bz2
osx-arm64/jupyter_core-4.7.0-py38h10201cd_0.tar.bz2        # maybe missing something?
osx-arm64/jupyter_core-4.7.0-py39h2804cbe_0.tar.bz2        # ?
win-64/jupyter_core-4.7.0-py36ha15d459_0.tar.bz2
win-64/jupyter_core-4.7.0-py37h03978a9_0.tar.bz2
win-64/jupyter_core-4.7.0-py38haa244fe_0.tar.bz2
win-64/jupyter_core-4.7.0-py39hcbf5309_0.tar.bz2
```

**Comment by bollwyvl** [comment]
I re-kicked the travis job for ppc:
https://travis-ci.com/github/conda-forge/jupyter_core-feedstock/builds/202572182

On osx-arm64, looks like we're just expecting 38/39 (whatever :shrug: )

**Comment by bollwyvl** [comment]
Was just a bad sort on my end: everything is fine. Thanks again @jasongrout !

**Pull request merged**
